RCT unpacker plus DQM 76X

nsmith- opened the pull request:
This is a 76X copy of #10381, without the `EventFilter/RctRawToDigi` history, which has already been merged in 76X (PR #10381)

mcepeda committed:
Add new code for rct unpacker, V0

mcepeda committed:
Changed source

mcepeda committed:
Now RCT&Layer2 unpacked plots

mcepeda committed:
Added central plots

Commit:
Two folder solution to RCT DQM

Commit:
Change folders to align with standalone dqm for RCT

Commit:
Updates to RawToDigi standard sequence for RctToDigi unpacker

Commit:
Remove rctDigis, too ambiguous

+ Delete old, very outdated files

Commit:
Change name of unpacker to rctHwDigis

Commit:
Adding requested changes from @jingyucms

Commit:
Introduce L1T PU monitoring module to online DQM

Commit:
Tweak positioning

Commit:
Change name of RCT unpacker to rctDigis in Standard Sequence

At the request of L1 offline software folks
For the record, I think this could possibly cause issues in
legacy code, due to the emulator previously being called 'rctDigis'
rather than the preferred 'simRctDigis'

cmsbuild commented:
A new Pull Request was created by @nsmith- (Nicholas Smith) for CMSSW_7_6_X.

RCT unpacker plus DQM 76X

It involves the following packages:

Configuration/StandardSequences
DQM/Integration
DQM/L1TMonitor


@cmsbuild, @danduggan, @franzoni, @deguio, @davidlange6 can you please review it and eventually sign? Thanks.
@ghellwig, @batinkov, @makortel, @GiacomoSguazzoni, @rovere, @VinInn, @Martin-Grunewald, @cerati, @threus, @dgulhan this is something you requested to watch as well.
You can sign-off by replying to this message having '+1' in the first line of your reply.
You can reject by replying  to this message having '-1' in the first line of your reply.
If you are a L2 or a release manager you can ask for tests by saying 'please test' in the first line of a comment.
@degano you are the release manager for this.
You can merge this pull request by typing 'merge' in the first line of your comment.

Commit:
Change default source of L1 RCT monitoring to caloStage1Digis

Commit:
Revert b603bf081b695ddd46f7b1c53f3046f6f4df6062

Due to merge conflict with 9a95bb204fd57e9bfcd0df681cf0344a0bafdf77

jingyucms commented:
@deguio Can you please test and review? The 74x version of this PR are running fine in dev machine.

cmsbuild commented:
Pull request #10658 was updated. @cmsbuild, @danduggan, @franzoni, @deguio, @davidlange6 can you please check and sign again.

deguio commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

deguio commented on DQM/L1TMonitor/src/L1TRCT.cc:
```diff
@@ -176,35 +213,63 @@ void L1TRCT::analyze(const Event & e, const EventSetup & c)
+  }
+
+
+  std::cout<<doHd<<doEm<<doHdLayer2<<doEmLayer2<<std::endl;
```
please remove cout

deguio commented:
-1
apart from the cout to be removed, the PR looks ok and can go in.
rejecting for now, but feel free to update this branch and the PR will be reconsidered.

Commit:
Remove cout

cmsbuild commented:
Pull request #10658 was updated. @cmsbuild, @danduggan, @franzoni, @deguio, @davidlange6 can you please check and sign again.

deguio commented:
please test

deguio commented:
+1

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-1 
Tested at: 85d793bac1356ce900fe1a70821bc5ec1f0612aa
 When I ran the RelVals I found an error in the following worklfows: 
 4.53 step3
<pre>runTheMatrix-results/4.53_RunPhoton2012B+RunPhoton2012B+HLTD+RECODreHLT+HARVESTDreHLT/step3_RunPhoton2012B+RunPhoton2012B+HLTD+RECODreHLT+HARVESTDreHLT.log

----- Begin Fatal Exception 14-Aug-2015 18:29:09 CEST-----------------------

An exception of category 'ProductNotFound' occurred while

   [0] Processing run: 194533 lumi: 329 event: 461582793

   [1] Running path 'dqmoffline_step'

   [2] Calling event method for module L1TdeRCT/'l1TdeRCT'

Exception Message:

Principal::getByToken: Found zero products matching all criteria

Looking for type: std::vector<L1CaloEmCand>

Looking for module label: caloStage1Digis

Looking for productInstanceName: 



   Additional Info:

      [a] If you wish to continue processing events after a ProductNotFound exception,

add "SkipEvent = cms.untracked.vstring('ProductNotFound')" to the "options" PSet in the configuration.



----- End Fatal Exception -------------------------------------------------

</pre>
140.53 step2
<pre>runTheMatrix-results/140.53_RunHI2011+RunHI2011+RECOHID11+HARVESTDHI/step2_RunHI2011+RunHI2011+RECOHID11+HARVESTDHI.log

----- Begin Fatal Exception 14-Aug-2015 18:32:04 CEST-----------------------

An exception of category 'ProductNotFound' occurred while

   [0] Processing run: 182124 lumi: 40 event: 1893722

   [1] Running path 'dqmoffline_step'

   [2] Calling event method for module L1TdeRCT/'l1TdeRCT'

Exception Message:

Principal::getByToken: Found zero products matching all criteria

Looking for type: std::vector<L1CaloEmCand>

Looking for module label: caloStage1Digis

Looking for productInstanceName: 



   Additional Info:

      [a] If you wish to continue processing events after a ProductNotFound exception,

add "SkipEvent = cms.untracked.vstring('ProductNotFound')" to the "options" PSet in the configuration.



----- End Fatal Exception -------------------------------------------------

</pre>
1000.0 step2
<pre>runTheMatrix-results/1000.0_RunMinBias2011A+RunMinBias2011A+TIER0+SKIMD+HARVESTDfst2+ALCASPLIT/step2_RunMinBias2011A+RunMinBias2011A+TIER0+SKIMD+HARVESTDfst2+ALCASPLIT.log

----- Begin Fatal Exception 14-Aug-2015 18:46:49 CEST-----------------------

An exception of category 'ProductNotFound' occurred while

   [0] Processing run: 165121 lumi: 62 event: 23609118

   [1] Running path 'dqmoffline_step'

   [2] Calling event method for module L1TdeRCT/'l1TdeRCT'

Exception Message:

Principal::getByToken: Found zero products matching all criteria

Looking for type: std::vector<L1CaloEmCand>

Looking for module label: caloStage1Digis

Looking for productInstanceName: 



   Additional Info:

      [a] If you wish to continue processing events after a ProductNotFound exception,

add "SkipEvent = cms.untracked.vstring('ProductNotFound')" to the "options" PSet in the configuration.



----- End Fatal Exception -------------------------------------------------

</pre>
1001.0 step2
<pre>runTheMatrix-results/1001.0_RunMinBias2011A+RunMinBias2011A+TIER0EXP+ALCAEXP+ALCAHARVD1+ALCAHARVD2+ALCAHARVD3/step2_RunMinBias2011A+RunMinBias2011A+TIER0EXP+ALCAEXP+ALCAHARVD1+ALCAHARVD2+ALCAHARVD3.log

----- Begin Fatal Exception 14-Aug-2015 18:51:28 CEST-----------------------

An exception of category 'ProductNotFound' occurred while

   [0] Processing run: 165121 lumi: 62 event: 23609118

   [1] Running path 'dqmoffline_step'

   [2] Calling event method for module L1TdeRCT/'l1TdeRCT'

Exception Message:

Principal::getByToken: Found zero products matching all criteria

Looking for type: std::vector<L1CaloEmCand>

Looking for module label: caloStage1Digis

Looking for productInstanceName: 



   Additional Info:

      [a] If you wish to continue processing events after a ProductNotFound exception,

add "SkipEvent = cms.untracked.vstring('ProductNotFound')" to the "options" PSet in the configuration.



----- End Fatal Exception -------------------------------------------------

</pre>
1003.0 step2
<pre>runTheMatrix-results/1003.0_RunMinBias2012A+RunMinBias2012A+RECODDQM+HARVESTDDQM/step2_RunMinBias2012A+RunMinBias2012A+RECODDQM+HARVESTDDQM.log

----- Begin Fatal Exception 14-Aug-2015 18:55:13 CEST-----------------------

An exception of category 'ProductNotFound' occurred while

   [0] Processing run: 191226 lumi: 398 event: 556864314

   [1] Running path 'dqmoffline_step'

   [2] Calling event method for module L1TdeRCT/'l1TdeRCT'

Exception Message:

Principal::getByToken: Found zero products matching all criteria

Looking for type: std::vector<L1CaloEmCand>

Looking for module label: caloStage1Digis

Looking for productInstanceName: 



   Additional Info:

      [a] If you wish to continue processing events after a ProductNotFound exception,

add "SkipEvent = cms.untracked.vstring('ProductNotFound')" to the "options" PSet in the configuration.



----- End Fatal Exception -------------------------------------------------

</pre>

 you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-10658/7305/summary.html

nsmith- commented:
@jingyucms do you know why the `caloStage1Digis` are not created in these tests?

jingyucms commented:
@nsmith caloStage1Digis is created. It fail because L1TdeRCT is looking for a std::vector format product but caloStage1Digis is actually a BxVector. You need a convertor to convert BxVector to std::vector to automatically feed the plots. And this is what I am doing.

nsmith- commented:
@jingyucms Are you sure?
Take the 1003.0 step2 RelVal for example, download the autogenerated config that is here:

https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-10658/7305/runTheMatrix-results/1003.0_RunMinBias2012A+RunMinBias2012A+RECODDQM+HARVESTDDQM/step2_RAW2DIGI_L1Reco_RECO_EI_DQM.py

And in the directory, try:
```
edmConfigDump step2_RAW2DIGI_L1Reco_RECO_EI_DQM.py > process.dump
grep 'caloStage1Digis' process.dump
grep 'L1TRawToDigi' process.dump
```
The only things I see when I do this are the L1TdeRCT modules trying to read in these digis.
`L1TRawToDigi` is the unpacker that creates the digis, and I won't expect it to work on a data file from 2012 as there was no stage 1 in 2012!

danduggan commented:
-1
still needs another commit before ready to go in

jingyucms commented:
@nsmith- can you run caloStage1Digis in l1TdeRCTSeq and l1TRCTSeq? ( L1TMonitor_cff.py and L1TEmulatorMonitor_cff.py)

nsmith- commented:
@jingyucms I think that unpacker steps should not belong in DQM-specific workflows, esp. when the unpackers for Stage-1 must certainly be in `StandardSequences` as we transition away from having a GCT.  Perhaps @mulhearn can elaborate.
@danduggan Can you expand your comment?  As far as I understand, this PR is being held back due to failing a RelVal that is impossible to pass.  Is it not possible to update the RelVals to reflect the fact that there are new FEDs, new unpackers, and therefore new event content for DQM in 2015 runs?

jingyucms commented:
@nsmith-  I think the falling RelVal can pass if you add the unpacker into your RCT sequence. Can you try?

jingyucms commented:
@nsmith- And add it to standard sequences will work also.

Commit:
Moving all rctDigis inside customization routines

DQM uses separate paths to differentiate stage1, so made
appropriate additions to stage1 paths

danduggan commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
Pull request #10658 was updated. @danduggan, @cmsbuild, @franzoni, @deguio, @mulhearn, @davidlange6 can you please check and sign again.

cmsbuild commented:
+1
Tested at: fd989c8354f3d2fdd94dbccc2ded00d52e9de3dd
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-10658/7479/summary.html

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-10658/7479/summary.html

nsmith- commented:
Superceded by #10882
